drivers/spi/spi_mcux_dspi.c

```diff
@@ -237,20 +250,240 @@ static const struct spi_driver_api spi_mcux_driver_api = {
 	.transceive = spi_mcux_transceive,
 };
 
+#else
+
+static void spi_mcux_transfer_next_packet(struct device *dev)
+{
+	const struct spi_mcux_config *config = dev->config->config_info;
+	struct spi_mcux_data *data = dev->driver_data;
+	SPI_Type *base = config->base;
+	struct spi_context *ctx = &data->ctx;
+	dspi_transfer_t transfer;
+
+	if ((ctx->tx_len == 0) && (ctx->rx_len == 0)) {
+		/* nothing left to rx or tx, we're done! */
+		spi_context_cs_control(&data->ctx, false);
+		spi_context_complete(&data->ctx, 0);
+		return;
+	}
+
+	if (ctx->tx_len == 0) {
+		/* rx only, nothing to tx */
+		transfer.txData = NULL;
+		transfer.rxData = ctx->rx_buf;
+		transfer.dataSize = ctx->rx_len;
+	} else if (ctx->rx_len == 0) {
+		/* tx only, nothing to rx */
+		transfer.txData = (u8_t *) ctx->tx_buf;
+		transfer.rxData = NULL;
+		transfer.dataSize = ctx->tx_len;
+	} else if (ctx->tx_len > ctx->rx_len) {
+		/* Break up the tx into multiple transfers so we don't have to
+		 * rx into a longer intermediate buffer.
+		 */
+		transfer.txData = (u8_t *) ctx->tx_buf;
+		transfer.rxData = ctx->rx_buf;
+		transfer.dataSize = ctx->rx_len;
+	} else {
+		/* Break up the rx into multiple transfers so we don't have to
+		 * tx from a longer intermediate buffer.
+		 */
+		transfer.txData = (u8_t *) ctx->tx_buf;
+		transfer.rxData = ctx->rx_buf;
+		transfer.dataSize = ctx->tx_len;
+	}
+
+	transfer.configFlags = kDSPI_MasterCtar0 | kDSPI_MasterPcsContinuous |
+			       (ctx->config->slave << DSPI_MASTER_PCS_SHIFT);
+
+	data->transfer_len = transfer.dataSize;
+
+	DSPI_MasterTransferNonBlocking(base, &data->handle, &transfer);
```

Comment:
Will this always be successful? Maybe some sys log message or an assertion could be useful here in case some unexpected circumstances occur?

Reply:
Yes it's possible this could fail. Will add a sys log or an assertion.